src/NativeRNDeviceInfoAndroid.ts

```diff
@@ -0,0 +1,154 @@
+import type { TurboModule } from 'react-native';
+import { TurboModuleRegistry } from 'react-native';
+
+export type LocationProviderInfo = {
+  [key: string]: boolean;
+};
+
+export type BatteryState = 'unknown' | 'unplugged' | 'charging' | 'full';
+
+export type PowerState = {
+  batteryLevel: number;
+  batteryState: BatteryState;
+  lowPowerMode: boolean;
+  [key: string]: string | number | boolean;
```

Comment:
Are there unlimited options that can also end up here?

Reply:
This is how it was typed already https://github.com/react-native-device-info/react-native-device-info/blob/98de4817c8911349023af5725f0a789dabb8641c/src/internal/types.ts#L5-L10 I didn't know the module well enough to make changes like that so kept what was already there.

Reply:
Ok so let's leave it like this right now and maybe update it in some follow-up PR.